lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp

```diff
@@ -1169,28 +1170,58 @@ void LowerClassesPass::lowerClass(om::ClassOp classOp, FModuleLike moduleLike,
     auto propertyResults = llvm::any_of(
         op.getResultTypes(), [](Type type) { return isa<PropertyType>(type); });
 
-    // If there are no properties here, move along.
-    if (!needsClone && !propertyOperands && !propertyResults)
+    return needsClone || propertyOperands || propertyResults;
+  };
+
+  // Move operations from the module like to the OM class.
+  // We need to clone only instances and register their results to the value
+  // mapping. Operands must be remapped using the value mapping.
+  for (auto &op : llvm::make_early_inc_range(
+           moduleLike->getRegion(0).front().getOperations())) {
+    if (!isPropertyOp(op))
       continue;
 
-    bool isField = false;
+    Operation *newOp = &op;
+
+    bool needsMove = true;
+
+    // Clone instances and register their results to the value mapping.
+    if (isa<InstanceOp>(op)) {
+      newOp = op.clone();
+      for (auto [oldResult, newResult] :
+           llvm::zip(op.getResults(), newOp->getResults()))
+        valueMapping[oldResult] = newResult;
+      classBody->getOperations().insert(classBody->end(), newOp);
+      needsMove = false;
+    }
+
+    // Remap operands using the value mapping.
+    for (size_t i = 0; i < newOp->getNumOperands(); ++i) {
+      auto operand = newOp->getOperand(i);
+      auto it = valueMapping.find(operand);
+      if (it != valueMapping.end())
+        newOp->setOperand(i, it->second);
+    }
+
+    // Move the op into the OM class body.
+    if (needsMove)
+      newOp->moveBefore(classBody, classBody->end());
+  }
```

Comment:
Simplification, maybe?

```suggestion
    // Clone instances and register their results to the value mapping.
    if (isa<InstanceOp>(op)) {
      newOp = op.clone();
      for (auto [oldResult, newResult] :
           llvm::zip(op.getResults(), newOp->getResults()))
        valueMapping[oldResult] = newResult;
      classBody->getOperations().insert(classBody->end(), newOp);
      continue;
    }

    // Remap operands using the value mapping.
    for (size_t i = 0; i < newOp->getNumOperands(); ++i) {
      auto operand = newOp->getOperand(i);
      auto it = valueMapping.find(operand);
      if (it != valueMapping.end())
        newOp->setOperand(i, it->second);
    }

    // Move the op into the OM class body.
    newOp->moveBefore(classBody, classBody->end());
  }
```